src/Features/Core/Portable/ConvertTupleToStruct/AbstractConvertTupleToStructCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,865 @@
+            if (addRenameAnnotation)
+            {
+                structNameToken = structNameToken.WithAdditionalAnnotations(RenameAnnotation.Create());
+            }
+
+            return typeParameters.Length == 0
+                ? (TNameSyntax)generator.IdentifierName(structNameToken)
+                : (TNameSyntax)generator.GenericName(structNameToken, typeParameters.Select(tp => generator.IdentifierName(tp.Name)));
+        }
+
+        private async Task<ImmutableArray<DocumentToUpdate>> GetDocumentsToUpdateAsync(
+            Document document, SyntaxNode tupleExprOrTypeNode,
+            INamedTypeSymbol tupleType, Scope scope, CancellationToken cancellationToken)
+        {
+            switch (scope)
+            {
+                case Scope.ContainingMember:
+                    return GetDocumentsToUpdateForContainingMember(document, tupleExprOrTypeNode);
+                case Scope.ContainingType:
+                    return await GetDocumentsToUpdateForContainingTypeAsync(
+                        document, tupleExprOrTypeNode, cancellationToken).ConfigureAwait(false);
+                case Scope.ContainingProject:
+                    return await GetDocumentsToUpdateForContainingProjectAsync(
+                        document.Project, tupleType, cancellationToken).ConfigureAwait(false);
+                case Scope.DependentProjects:
+                    return await GetDocumentsToUpdateForDependentProjectAsync(
+                        document.Project, tupleType, cancellationToken).ConfigureAwait(false);
+                default:
+                    break;
+            }
+
+            throw new NotImplementedException();
+        }
+
+        private async Task<ImmutableArray<DocumentToUpdate>> GetDocumentsToUpdateForDependentProjectAsync(
+            Project startingProject, INamedTypeSymbol tupleType, CancellationToken cancellationToken)
+        {
+            var solution = startingProject.Solution;
+            var graph = solution.GetProjectDependencyGraph();
+
+            // Note: there are a couple of approaches we can take here.  Processing 'direct'
+            // dependencies, or processing 'transitive' dependencies.  Both have pros/cons:
+            //
+            // Direct Dependencies:
+            //  Pros:
+            //      All updated projects are able to see the newly added type.
+            //      Transitive deps won't be updated to use a type they can't actually use.
+            //  Cons:
+            //      If that project then exports that new type, then transitive deps will
+            //      break if they use those exported APIs since they won't know about the
+            //      type.
+            //
+            // Transitive Dependencies:
+            //  Pros:
+            //      All affected code is updated.
+            //  Cons: 
+            //      Non-direct deps will not compile unless the take a reference on the
+            //      starting project.
```

Comment:
>// [](start = 12, length = 2)

The comment should say which approach is actually taken ;-)

Reply:
Please clarify this comment to say which approach is taken.

---
In reply to: [201863419](https://github.com/dotnet/roslyn/pull/28257#discussion_r201863419) [](ancestors = 201863419)